packages/klighd-core/src/filtering/semantic-filtering-util.ts

```diff
@@ -0,0 +1,318 @@
+    static NAME = "NOT"
+    name = NegationConnective.NAME
+    operand: SemanticFilterRule
+    ruleName?: string
+}
+
+/**
+ * An And Connective takes two rules R1 and R2 and evaluates to true
+ * iff
+ * R1 and R2 evaluate to true.
+ * @example R1 && R2
+ */
+export class AndConnective implements BinaryConnective {
+    static NAME = "AND"
+    name = AndConnective.NAME
+    leftOperand: SemanticFilterRule
+    rightOperand: SemanticFilterRule
+    ruleName?: string
+}
+
+/**
+ * An Or Connective takes two rules R1 and R2 and evaluates to true
+ * iff
+ * R1 or R2 evaluate to true.
+ * @example R1 || R2
+ */
+export class OrConnective implements BinaryConnective {
+    static NAME = "OR"
+    name = OrConnective.NAME
+    leftOperand: SemanticFilterRule
+    rightOperand: SemanticFilterRule
+    ruleName?: string
+}
+
+/**
+ * An IfThen Connective takes two rules R1 and R2 and evaluates to true
+ * iff
+ * R1 evaluates to false or R2 evaluates to true.
+ * @example R1 ? R2 : true
+ * @example !R1 || R2
+ */
+export class IfThenConnective implements BinaryConnective {
+    static NAME = "IFTHEN"
+    name = IfThenConnective.NAME
+    leftOperand: SemanticFilterRule
+    rightOperand: SemanticFilterRule
+    ruleName?: string
+}
+
+/**
+ * A LogicEqual Connective takes two rules R1 and R2 and evaluates to true
+ * iff
+ * R1 and R2 evaluate to true or R1 and R2 evaluate to false.
+ * @example R1 === R2
+ * @example R1 && R2 || !R1 && !R2
+ */
+export class LogicEqualConnective implements BinaryConnective {
+    static NAME = "LOGICEQUAL"
+    name = LogicEqualConnective.NAME
+    leftOperand: SemanticFilterRule
+    rightOperand: SemanticFilterRule
+    ruleName?: string
+}
+
+/**
+ * An IfThenElse Connective takes three rules R1, R2 and R3 and evaluates to true
+ * iff
+ * R1 and R2 evaluate to true or R1 evaluates to false and R3 evaluates to true.
+ * @example R1 ? R2 : R3
+ * @example R1 && R2 || !R1 && R3
+ */
+export class IfThenElseConnective implements TernaryConnective {
+    static NAME = "IFTHENELSE"
+    name = IfThenElseConnective.NAME
+    firstOperand: SemanticFilterRule
+    secondOperand: SemanticFilterRule
+    thirdOperand: SemanticFilterRule
+    ruleName?: string
+}
+
+//// Numeric Connectives ////
+
+/**
+ * A LessThan Connective takes one rule R and evaluates to true
+ * iff
+ * R.num < correspondingTag.num.
+ * @example R.num < correspondingTag.num
+ */
+export class LessThanConnective implements UnaryConnective {
+    static NAME = "LESSTHAN"
+    name = LessThanConnective.NAME
+    operand: SemanticFilterTag
+    ruleName?: string
+}
+
+/**
+ * A GreaterThan Connective takes one rule R and evaluates to true
+ * iff
+ * R.num > correspondingTag.num.
+ * @example R.num > correspondingTag.num
+ */
+export class GreaterThanConnective implements UnaryConnective {
+    static NAME = "GREATERTHAN"
+    name = GreaterThanConnective.NAME
+    operand: SemanticFilterTag
+    ruleName?: string
+}
```

Comment:
Default relation definitions usually is using `<=`, `>=` and `=` resp. `!=` instead of `<` and `>`